packages/roosterjs-editor-plugins/lib/plugins/TableCellSelection/utils/forEachSelectedCell.ts

```diff
@@ -10,18 +10,13 @@ import { VTable } from 'roosterjs-editor-dom';
  */
 export function forEachSelectedCell(vTable: VTable, callback: (cell: VCell) => void): number {
     let selectedCells = 0;
+    vTable.selection = normalizeTableSelection(vTable.selection, vTable.table);
 
     const { lastCell, firstCell } = vTable.selection;
 
     for (let y = 0; y < vTable.cells.length; y++) {
```

Comment:
then why not just
for (i from selecton.row1 to row2)
  for (j from selection.col1 to col2) ...